src/app/log.cpp

```diff
@@ -0,0 +1,28 @@
+#include <app/log.hpp>
+
+Log::Config Log::_config;
+std::ofstream Log::log_out_fs;
+
+void Log::init() {
+    log_out_fs.open(_config.log_filename, std::ios_base::app);
+}
+
+void Log::initFromConfig(const Log::Config &init_config) {
+    _config = init_config;
+    init();
+}
+void Log::initFromConfig(const std::string &filename) {
+    json::load<Config>(filename, _config);
+    init();
+}
```

Comment:
Not sure if we need this overload since you can always use `initFromConfig(json::load<Config>(filename))`
This also creates an additional dependency on JSON

Reply:
It already has dependencies on JSON due to JSON_MAPPINGS at least

Reply:
Deleted